client/proto/request/v2/request.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 VMware, all rights reserved
+//
+// Concordclient's request service
+
+syntax = "proto3";
+package vmware.concord.client.request.v2;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/any.proto";
+
+// Specifies Java package name, using the standard prefix "com."
+option java_package = "com.vmware.concord.client.request.v2";
+
+// Service error handling
+// Each service method will return a gRPC status object.
+// All errors are mapped to the common gRPC error codes defined here: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+// See each method for concrete explanations below.
+
+// The RequestService allows the caller to submit requests to the blockchain
+// network, and receive the result of the request in a synchronous response.
+//
+// You can use the EventService below for asynchronously consuming the
+// events of submitted requests.
+service RequestService {
+  // Send a single request via the Concord Client to the blockchain network.
+  // Errors:
+  // DEADLINE_EXCEEDED: if the request couldn't be processed before the given timeout expired.
+  // INVALID_ARGUMENT: if a required field is not set.
+  // RESOURCE_EXHAUSTED: if Concord Client is overloaded. The caller should retry with a backoff.
+  // UNAVAILABLE: if Concord Client is currently unable to process any requests. The caller should retry with a backoff.
+  // ABORTED: if Concord has a contention between concurrently running requests. The caller should retry.
+  // INTERNAL: if Concord Client cannot progress independent of the request.
+  rpc Send(Request) returns (Response);
+}
+
+message Request {
+  // Required application request which gets evaluated by the execution engine.
+  bytes request = 1;
```

Comment:
We can still [keep wire compatibility](https://developers.google.com/protocol-buffers/docs/proto3#backwards-compatibility_issues) by introducing a `oneof` for the typed vs. non-typed request:
```suggestion
  oneof application_request {
    bytes raw_request = 1 [deprecated = true];
    google.protobuf.Any typed_request = 6;
  }
```
The same may be applied to the `Response` message. I.e., this way we make it explicit that only one of these fields will be ever set.
I have annotated the old field as deprecated above as well (in a subsequent service version it may be removed).

Reply:
This could be tricky which is the reason why OneOf was not introduced. The document does mention about backward compatibility but that would be when we are moving existing fields to a new OneOf. But here we are creating a new Oneof and also introducing a new field typed_request  into the OneOf which  is not the same case.
cc: @robem

Reply:
@cjn Introducing a new `oneof` that encapsulates an existing and a new field has the same wire-compatibility guarantees as moving more than one existing fields into a `oneof`. I.e., old clients that do not know about the new field just won't be able to interpret requests that have the `oneof` field populated that they are not aware of.
@robem I suggested using `oneof` + `[deprecated = true]` exactly for this reason, i.e., this way we don't have to introduce a new service version as the suggested changes are wire-compatible.

Reply:
Thanks @miklos-da  and @robem. I will take these inputs into consideration for existing service.

Reply:
Updated changes to PR